### PR TITLE
add config rule support to policylambda.py

### PR DESCRIPTION
**Pull request by CYarros10**
Solution for Issue #6051

**Is your feature request related to a problem? Please describe.**
When using cloud-custodian/tools/ops/policylambda.py to convert custodian policy documents into cloudformation templates, there is missing functionality for the modes config-rule and config-poll-rule.

**Describe the solution you'd like**
We would like cloud-custodian/tools/ops/policylambda.py to add support for modes config-rule and config-poll-rule to generate SAM/Cloudformation for Config Rules. We need to convert custodian policies containing AWS Config Rules into cloudformation so that underlying resources can be linted and managed.

### CODE REVIEW

**Comment by kapilt** [comment]
followup from #6066 

Thanks for the pull request!  If you don't mind could you sign the CLA per the developer docs? Direct link here: https://docs.google.com/forms/d/e/1FAIpQLSfwtl1s6KmpLhCY6CjiY8nFZshDwf_wrmNYx1ahpsNFXXmHKw/viewform

we're in process on switching out to the cncf cla but thats a few weeks out.

File: tools/ops/policylambda.py [hunk 1]
```diff
@@ -54,36 +56,81 @@ def render(p):
...
-                    'Type': 'CloudWatchEvent',
+        if p.execution_mode == 'periodic':
+            revents = {
+                'PolicySchedule': {
+                    'Type': 'Schedule',
                     'Properties': {
-                        'Pattern': json.loads(e.render_event_pattern())}
+                        'Schedule': p.data.get('mode', {}).get('schedule')}}
             }
+        else:
```
Comment (kapilt) [inline tools/ops/policylambda.py:69]: thanks for the followup pr, this block should be dedented to an if == 'cloudtrail' with the additional clause.

Comment (CYarros10) [reply tools/ops/policylambda.py]: updated based on feedback

**Comment by CYarros10** [comment]
> followup from #6066
> 
> Thanks for the pull request! If you don't mind could you sign the CLA per the developer docs? Direct link here: https://docs.google.com/forms/d/e/1FAIpQLSfwtl1s6KmpLhCY6CjiY8nFZshDwf_wrmNYx1ahpsNFXXmHKw/viewform
> 
> we're in process on switching out to the cncf cla but thats a few weeks out.

signed!

File: tools/ops/policylambda.py [hunk 2]
```diff
@@ -77,13 +81,56 @@ def render(p):
...
+        "Properties": {
+            "Action": "lambda:InvokeFunction",
+            "FunctionName": {"Ref": name},
+            "Principal": "config.amazonaws.com"
+        }
+    }
+
+
+def get_lambda_arn(function_name):
+    sts = boto3.client("sts")
```
Comment (kapilt) [inline tools/ops/policylambda.py:125]: we don't really want the arn, we want a cfn reference to the function. we should update mu.py here to support pass through on the reference.

File: tools/ops/policylambda.py [hunk 3]
```diff
@@ -77,13 +81,56 @@ def render(p):
...
+        properties.pop('Scope', None)
+
+    attributes = {}
+    attributes['Type'] = 'AWS::Config::ConfigRule'
+    attributes['DependsOn'] = resource_name(p.name) + "Invoke"
+    attributes['Properties'] = properties
+    return attributes
+
+
+def render_invoke(name):
```
Comment (kapilt) [inline tools/ops/policylambda.py:112]: so sam generally takes care of this for cwe, let's rename this function to account for its specificity to config

**Pull request merged**